usr.sbin/bhyve/amd64/pci_irq_md.h

```diff
@@ -0,0 +1,59 @@
+/*-
+ * SPDX-License-Identifier: BSD-2-Clause
+ *
+ * Copyright (c) 2014 Hudson River Trading LLC
+ * Written by: John H. Baldwin <jhb@FreeBSD.org>
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE AUTHOR AND CONTRIBUTORS ``AS IS'' AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED.  IN NO EVENT SHALL THE AUTHOR OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS
+ * OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION)
+ * HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY
+ * OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF
+ * SUCH DAMAGE.
+ */
+
+#ifndef __PCI_IRQ_MD_H__
+#define	__PCI_IRQ_MD_H__
+
+struct vmctx;
+
+struct pci_irq {
+	int	pirq_pin;
+	int	ioapic_irq;
+};
+
+void	pci_irq_init(struct vmctx *ctx);
+void	pci_irq_reserve(int irq);
+void	pci_irq_use(int irq);
+uint8_t	pirq_read(int pin);
+void	pirq_write(struct vmctx *ctx, int pin, uint8_t val);
+
+static inline void
+pci_irq_init_irq(struct pci_irq *irq)
+{
+	irq->pirq_pin = 0;
+	irq->ioapic_irq = 0;
+}
+
+static inline uint8_t
+pci_irq_intline(struct pci_irq *irq)
+{
+	return (pirq_irq(irq->pirq_pin));
```

Comment:
Hmm, I can't find the implementation of `pirq_irq` in this patch for amd64 now, seems like this wouldn't compile on amd64?

Reply:
(And yet it compiles in CI so must be there somewhere?)

Reply:
Hmmm not sure what's going on there

Reply:
```        if self.crosscompile_target.is_any_x86():
            # seems to be missing some include paths which appears to work on freebsd
            self.make_args.set_with_options(BHYVE=False)
``` :(

Reply:
Yeah, assuming that's the only thing I broke

Reply:
The updated version worked for me on my amd64 laptop, I tested an i386 VM with MSI disabled (so using INTx interrupts via the I/O APIC) for ahci0, and also with APIC disabled entirely, so using the PIR and 8259A PICs.